autoload/lsp/completion.vim

```diff
@@ -425,16 +425,23 @@ enddef
 export def BufferInit(lspserver: dict<any>, bnr: number, ftype: string)
   # set options for insert mode completion
   if opt.lspOptions.autoComplete
+    var noSel = opt.lspOptions.noSelInCompletion ? ',noselect' : ''
+
     if lspserver.completionLazyDoc
-      setbufvar(bnr, '&completeopt', 'menuone,popuphidden,noinsert,noselect')
+      setbufvar(bnr, '&completeopt', 'menuone,popuphidden,noinsert' .. noSel)
       setbufvar(bnr, '&completepopup', 'width:80,highlight:Pmenu,align:item,border:off')
     else
-      setbufvar(bnr, '&completeopt', 'menuone,popup,noinsert,noselect')
+      setbufvar(bnr, '&completeopt', 'menuone,popup,noinsert' .. noSel)
       setbufvar(bnr, '&completepopup', 'border:off')
     endif
     # <Enter> in insert mode stops completion and inserts a <Enter>
     if !opt.lspOptions.noNewlineInCompletion
       inoremap <expr> <buffer> <CR> pumvisible() ? "\<C-Y>\<CR>" : "\<CR>"
+    endif
+
+	# <Tab> in insert mode confirms selected item in completion list
+    if opt.lspOptions.useTabInCompletion
+      inoremap <expr> <buffer> <Tab> pumvisible() ? "\<C-Y>" : "\<Tab>"
```

Comment:
I think that these two completions should be unified to one option, possible:

```
lsp.completionTrigger = "cr" | "tab" | ""
```

Which would cause these two keybinds:

    if opt.lsp.completionTrigger == "cr"
      inoremap <expr> <buffer> <CR> pumvisible() ? "\<C-Y>\<CR>" : "\<CR>"
    elseif opt.lsp.completionTrigger == "tab"
      inoremap <expr> <buffer> <Tab> pumvisible() ? "\<C-Y>" : "\<Tab>"
    endif
    
Where `lspOptions.noNewlineInCompletion = true` would be deprecated in favor of  `opt.lsp.completionTrigger = ''` and the default would be `opt.lsp.completionTrigger = 'cr'`.